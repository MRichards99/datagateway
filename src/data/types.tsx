export interface InvestigationData {
  ID: string;
  TITLE: string;
  VISIT_ID: number;
  RB_NUMBER: string;
  DOI: string;
  SIZE: number;
  INSTRUMENT: { NAME: string };
  STARTDATE: Date;
  ENDDATE: Date;
  [key: string]: string | number | Date | { NAME: string };
}

export interface DatasetData {
  ID: string;
  NAME: string;
  SIZE: number;
  CREATE_TIME: Date;
  MOD_TIME: Date;
  [key: string]: string | number | Date;
}

export interface DatafileData {
  ID: string;
  NAME: string;
  LOCATION: string;
  SIZE: number;
  MOD_TIME: Date;
  [key: string]: string | number | Date;
}

export interface FoodData {
  id: number;
  dessert: string;
  calories: number;
  fat: number;
  carbs: number;
  protein: number;
  [key: string]: string | number;
}

export type EntityType =
  | InvestigationData
  | DatasetData
  | DatafileData
  | FoodData;
