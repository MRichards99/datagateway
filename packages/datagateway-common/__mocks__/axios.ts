import axios from 'axios';

export default {
  get: jest.fn(() => Promise.resolve({ data: {} })),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  delete: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};