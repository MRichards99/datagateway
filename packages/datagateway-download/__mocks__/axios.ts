import axios from 'axios';

export default {
  get: jest.fn(path => {
    if (path.includes('/topcat/user/cart/')) {
      return Promise.resolve({ data: { cartItems: [] } });
    } else if (path === '/datagateway-download-settings.json') {
      return Promise.resolve({
        data: {
          facilityName: 'LILS',
          apiUrl: 'http://scigateway-preprod.esc.rl.ac.uk:5000',
          downloadApiUrl: 'https://scigateway-preprod.esc.rl.ac.uk:8181/topcat',
          idsUrl: 'https://scigateway-preprod.esc.rl.ac.uk:8181/ids',
          accessMethods: {
            https: {
              idsUrl: 'https://scigateway-preprod.esc.rl.ac.uk:8181/ids',
              displayName: 'HTTPS',
              description: 'Example description for HTTPS access method.',
            },
            globus: {
              idsUrl: 'https://scigateway-preprod.esc.rl.ac.uk:8181/ids',
              displayName: 'Globus',
              description: 'Example description for Globus access method.',
            },
          },
        },
      });
    } else {
      return Promise.resolve({ data: {} });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};
