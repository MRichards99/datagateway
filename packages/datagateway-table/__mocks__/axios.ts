import axios from 'axios';

export default {
  get: jest.fn(path => {
    if (path === '/settings.json') {
      return Promise.resolve({
        data: {
          'ui-strings': '/res/default.json',
        },
      });
    } else {
      return Promise.resolve({
        data: {},
      });
    }
  }),
  post: jest.fn(() => Promise.resolve({ data: {} })),
  CancelToken: axios.CancelToken,
};
