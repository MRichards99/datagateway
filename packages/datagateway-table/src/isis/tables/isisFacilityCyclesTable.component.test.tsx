import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import ISISFacilityCyclesTable from './isisFacilityCyclesTable.component';
import { initialState } from '../../state/reducers/dgtable.reducer';
import configureStore from 'redux-mock-store';
import { StateType } from '../../state/app.types';
import {
  fetchFacilityCyclesRequest,
  filterTable,
  sortTable,
} from '../../state/actions';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { TableSortLabel } from '@material-ui/core';
import { MemoryRouter } from 'react-router';
import axios from 'axios';

describe('ISIS FacilityCycles table component', () => {
  let shallow;
  let mount;
  let mockStore;
  let state: StateType;
  (axios.get as jest.Mock).mockImplementation(() =>
    Promise.resolve({ data: [] })
  );

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'div' });
    mount = createMount();

    mockStore = configureStore([thunk]);
    state = JSON.parse(JSON.stringify({ dgtable: initialState }));
    state.dgtable.data = [
      {
        ID: 1,
        NAME: 'Test 1',
        DESCRIPTION: 'Test 1',
        STARTDATE: '2019-07-03',
        ENDDATE: '2019-07-04',
      },
    ];
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <ISISFacilityCyclesTable store={mockStore(state)} instrumentId="1" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('sends fetchFacilityCycles action on load', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISFacilityCyclesTable instrumentId="1" />
        </MemoryRouter>
      </Provider>
    );

    expect(testStore.getActions()[0]).toEqual(fetchFacilityCyclesRequest());
  });

  it('sends filterTable action on filter', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISFacilityCyclesTable instrumentId="1" />
        </MemoryRouter>
      </Provider>
    );

    const filterInput = wrapper.find('input').first();
    filterInput.instance().value = 'test';
    filterInput.simulate('change');

    expect(testStore.getActions()[1]).toEqual(filterTable('NAME', 'test'));

    filterInput.instance().value = '';
    filterInput.simulate('change');

    expect(testStore.getActions()[2]).toEqual(filterTable('NAME', null));
  });

  it('sends sortTable action on sort', () => {
    const testStore = mockStore(state);
    const wrapper = mount(
      <Provider store={testStore}>
        <MemoryRouter>
          <ISISFacilityCyclesTable instrumentId="1" />
        </MemoryRouter>
      </Provider>
    );

    wrapper
      .find(TableSortLabel)
      .first()
      .simulate('click');

    expect(testStore.getActions()[1]).toEqual(sortTable('NAME', 'asc'));
  });

  it('renders facilitycycle name as a link', () => {
    const wrapper = mount(
      <Provider store={mockStore(state)}>
        <MemoryRouter>
          <ISISFacilityCyclesTable instrumentId="1" />
        </MemoryRouter>
      </Provider>
    );

    expect(
      wrapper
        .find('[aria-colindex=2]')
        .find('p')
        .children()
    ).toMatchSnapshot();
  });
});
