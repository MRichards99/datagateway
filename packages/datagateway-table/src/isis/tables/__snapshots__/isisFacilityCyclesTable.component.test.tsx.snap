// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS FacilityCycles table component renders correctly 1`] = `
<div
  style={
    Object {
      "height": 0,
      "overflow": "visible",
      "width": 0,
    }
  }
>
  <InfiniteLoader
    isRowLoaded={[Function]}
    loadMoreRows={[Function]}
    minimumBatchSize={25}
    rowCount={1}
    threshold={15}
  >
    <Component />
  </InfiniteLoader>
</div>
`;

exports[`ISIS FacilityCycles table component renders facilitycycle name as a link 1`] = `null`;
